def/babel6.js

```diff
@@ -0,0 +1,125 @@
+        .field("directives", [def("Directive")], defaults.emptyArray);
+
+    // Split Literal
+    def("StringLiteral")
+        .bases("Expression")
+        .build("value")
+        .field("value", String);
+
+    def("NumericLiteral")
+        .bases("Expression")
+        .build("value")
+        .field("value", Number);
+
+    def("NullLiteral")
+        .bases("Expression")
+        .build();
+
+    def("BooleanLiteral")
+        .bases("Expression")
+        .build("value")
+        .field("value", Boolean);
+
+    def("RegExpLiteral")
+        .bases("Expression")
+        .build("pattern", "flags")
+        .field("pattern", String)
+        .field("flags", String);
+
+    // Split Property -> ObjectProperty and ObjectMethod
+    def("ObjectExpression")
+        .bases("Expression")
+        .build("properties")
+        .field("properties", [or(def("ObjectMethod"), def("ObjectProperty"), def("SpreadProperty"))]);
+
+    // ObjectMethod hoist .value properties to own properties
+    def("ObjectMethod")
+        .bases("Node")
+        .build("kind", "key", "params", "body", "computed")
+        .field("kind", or("method", "get", "set"))
+        .field("key", or(def("Literal"), def("Identifier"), def("Expression")))
+        .field("params", [def("Pattern")])
+        .field("body", def("BlockStatement"))
+        .field("computed", Boolean, defaults["false"])
+        .field("generator", Boolean, defaults["false"])
+        .field("async", Boolean, defaults["false"])
+        .field("decorators",
+            or([def("Decorator")], null),
+            defaults["null"]);
+
+    def("ObjectProperty")
+        .bases("Node")
+        .build("key", "value")
+        .field("key", or(def("Literal"), def("Identifier"), def("Expression")))
+        .field("value", or(def("Expression"), def("Pattern")))
+        .field("computed", Boolean, defaults["false"]);
+
+    // MethodDefinition -> ClassMethod
+    def("ClassBody")
+        .bases("Declaration")
+        .build("body")
+        .field("body", [or(def("ClassMethod"), def("ClassProperty"))]);
+
+    def("ClassMethod")
+        .bases("Declaration")
+        .build("kind", "key", "params", "body", "computed", "static")
+        .field("kind", or("get", "set", "method", "constructor"))
+        .field("key", or(def("Literal"), def("Identifier"), def("Expression")))
+        .field("params", [def("Pattern")])
+        .field("body", def("BlockStatement"))
+        .field("computed", Boolean, defaults["false"])
+        .field("static", Boolean, defaults["false"])
+        .field("generator", Boolean, defaults["false"])
+        .field("async", Boolean, defaults["false"])
+        .field("decorators",
+            or([def("Decorator")], null),
+            defaults["null"]);
+
+    // Split into RestProperty and SpreadProperty
+    def("ObjectPattern")
+        .bases("Pattern")
+        .build("properties")
+        .field("properties", [or(def("RestProperty"), def("ObjectProperty"))])
```

Comment:
Note how ObjectPattern.properties is an array of ObjectProperty's, like you have, but ObjectProperty's type field is "Property", not "ObjectProperty." def("ObjectProperty") requires an ast with { type: "ObjectProperty" }, but estree says it should be { type: "Property" }